docs/content/en/docs/tutorials/ci_cd.md

```diff
@@ -0,0 +1,286 @@
+              - ./deployment/dev/secrets.yaml
+            skipBuildDependencies: true
+            artifactOverrides:
+              image: asia.gcr.io/my-project/my-image
+            imageStrategy:
+              helm: {}
+        flags:
+          upgrade:
+            - --install
+```
+
+
+Here, we are
+
+
+
+*   **Creating a profile** - You can set up multiple profiles using skaffold and deploy them separately if you want using **skaffold run -p &lt;profilename>**
+  
+*   **Specifying the artifacts to build from** - In this case, we are asking skaffold to build from a Dockerfile, passing its args and also specifying the destination where the image needs to be pushed. The values can also come from the env (NPM_REGISTRY and NPM_TOKEN in this example)
+
+*   **Specifying the deployment strategy** - In this case, helm is specified with the location of the chart, all the configuration as documented in the yaml reference and also the values to be passed to it. We are also using the helm secrets plugin above.
+
+
+If you want to deploy via kubectl, your skaffold file can look something like this:
+
+
+```
+apiVersion: skaffold/v2beta8
+kind: Config
+profiles:
+    - name: dev-svc
+      build:
+          artifacts:
+              - image: asia.gcr.io/my-project/my-image
+                docker:
+                    buildArgs:
+                        NPM_REGISTRY: '{{.NPM_REGISTRY}}'
+                        NPM_TOKEN: '{{.NPM_TOKEN}}'
+      deploy:
+          statusCheckDeadlineSeconds: 600
+          kubectl:
+              manifests:
+                  - k8/dev/*.yml
+```
+
+
+**Please refer to the yaml reference for all that you can specify in skaffold.yaml file**
+
+
+## Step 6: Development & Testing locally
+
+Before we move on to setting up the CI pipeline, we should test it out locally. Use **skaffold run** if you are deploying it to the cluster, **skaffold build **to just build and push the artifacts, **skaffold dev **if you are developing (this will enable auto reload on changes) or debug using **skaffold debug**
+
+You can find docs about all these workflows here: [https://skaffold.dev/docs/workflows/](https://skaffold.dev/docs/workflows/)
+
+Also, note that you can also set up file synchronization to enable faster development workflows. You can read more about that here: [https://skaffold.dev/docs/pipeline-stages/filesync/](https://skaffold.dev/docs/pipeline-stages/filesync/)
+
+
+## Step 7: Getting the CI pipeline setup
+
+Now that we have skaffold ready and working locally, the CI pipeline requires little to no effort since it will just call skaffold to do all the builds and deployments.
+
+Here is a sample pipeline using Gitlab CI
+
+
+```
+stages:
+    - my-stage
+
+my-job:
+    image:
+        name: gcr.io/k8s-skaffold/skaffold:v1.15.0
+    tags:
+        - development
+    stage: my-stage
+    retry: 2
+    script:
+        - docker login $REG_DEV_URL -u $REG_DEV_UNAME -p $REG_DEV_PASSWORD
```

Comment:
@briandealwis Actually, I used to have a private registry (Harbor) and this is how I used to authenticate to it. Since this is going to be a public tutorial, I replaced the image URL with gcr but then left this instruction out as such. This is for passing in user name, password and registry url in case you are either using docker hub or a private registry.

Should I change this to make it suitable to GCR? The process can be different for different registry provider, I thought this would be consistent. Let me know.